tensorflow_datasets/translate/en_tam_parallel_text.py

```diff
@@ -0,0 +1,108 @@
+"""English-Tamil parallel text corpus"""
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import os
+import tensorflow as tf
+import tensorflow_datasets as tfds
+
+
+_CITATION = """
+@inproceedings {biblio:RaBoMorphologicalProcessing2012,
+	title = {Morphological Processing for English-Tamil Statistical Machine Translation},
+	author = {Loganathan Ramasamy and Ond{\v{r}}ej Bojar and Zden{\v{e}}k {\v{Z}}abokrtsk{\'{y}}},
+	year = {2012},
+	pages = {113--122},
+	Booktitle = {Proceedings of the Workshop on Machine Translation and Parsing in Indian Languages ({MTPIL}-2012)},
+}
+
+@InProceedings{TIEDEMANN12.463,
+  author = {J�rg Tiedemann},
+  title = {Parallel Data, Tools and Interfaces in OPUS},
+  booktitle = {Proceedings of the Eight International Conference on Language Resources and Evaluation (LREC'12)},
+  year = {2012},
+  month = {may},
+  date = {23-25},
+  address = {Istanbul, Turkey},
+  editor = {Nicoletta Calzolari (Conference Chair) and Khalid Choukri and Thierry Declerck and Mehmet Ugur Dogan and Bente Maegaard and Joseph Mariani and Jan Odijk and Stelios Piperidis},
+  publisher = {European Language Resources Association (ELRA)},
+  isbn = {978-2-9517408-7-7},
+  language = {english}
+ }
+"""
+_DESCRIPTION = """\
+This corpus is a collection of english and tamil parallel text extracted 3 sources.
+All the three were merged into a single file and cleaned.The various data cleaning steps were
+a)Removed duplicates lines
+b)Removed lines containing html tags
+c) Removed empty lines and lines which contains an empty english or tamil translation
+d) Manually inspected and removed junk characters
+There are total of 370,000 (roughly) lines of text which were split to train, validation and test sets. 
+Script used for data cleaning :- https://colab.research.google.com/drive/1AViiS_4-ClngwZ4bKYbmlq3cVAnNHCk0#scrollTo=MWn-AuohbfXI
+
+Note:-
+For dataset-3, the data was created by non-expert translators hired over Mechanical Turk and the other 
+datasets were created from extracting text from movie subtitles, Bible, Quran, cinema and news websites so the quality of translations are of mixed.
+However, it should be useful enough to get you started training models.
+
+Reference for dataset 1 :- http://ufal.mff.cuni.cz/~ramasamy/parallel/html/
+Reference for dataset 2 :- http://opus.nlpl.eu/
+Reference for dataset 3 :- https://github.com/joshua-decoder/indian-parallel-corpora
+License   for dataset 3 :- https://github.com/joshua-decoder/indian-parallel-corpora/blob/master/LICENSE
+"""
+test_file_link = 'https://github.com/praveenjune17/Neural-Machine-Translation-English-Tamil-model/raw/master/en_tam_parallel_text_test.tar.gz'
+valid_file_link = 'https://github.com/praveenjune17/Neural-Machine-Translation-English-Tamil-model/raw/master/en_tam_parallel_text_valid.tar.gz'
```

Comment:
Has the dataset any original test and valid files? Thanks for thinking and adding but you must need to give original files links.

Reply:
Except the 3rd dataset (joshua-decoder) the other datasets didn't have any original test or valid files. Mentioned the original file links in "en_tam_parallel_text.py"( in  the description) , I collected the data from these sources before I even know about TFDS and stored in my local PC. Even after doing some preprocessing to the best of knowledge there were many manual corrections done in the dataset to keep the noise low.  If the original files links are still needed let me know I'll figure out a way to do it.

Thanks !

Reply:
Yes, but the dataset has to be the original link for tfds. 
@rsepassi @Conchylicultor @cyfra what do you think about that?

Reply:
https://github.com/praveenjune17/Neural-Machine-Translation-English-Tamil-model/blob/master/Transformer_en_tam.ipynb

Please have a look at the above machine translation script. I have patched the en_tam scripts into the /usr/local.../translate directory and trained the transformer by loading the data using tfds. It works like a gem , the translation output is also good. As I already mentioned , I have done many manual corrections in the dataset to keep the noise low and I'm not sure whether I would get the same quality of translations without those manual corrections .

Reply:
Yes, but it is better if you download the files from the original source and make corrections to the tfds dataset file.

Reply:
sure..started working on that